lib/src/gemm_blocked_rvm.c

```diff
@@ -13,8 +17,33 @@
  * @param m Number of columns in matrix A and number of rows in matrix B.
  * @param k Number of columns in matrix B and resulting matrix C.
  */
-extern void gemm_block4x4_rvm(const float *A, const float *B, float *C, const size_t n, const size_t m, const size_t k)
-{
-    // TODO: Implement Gemm using THEAD RISC-V matrix extension.
-    gemm_ref(A, B, C, n, m, k);    
+extern void gemm_block4x4_rvm(const float *A, const float *B, float *C, const size_t n, const size_t m, const size_t k) {
+    mcfgm(BLOCK_SIZE);
+    mcfgn(BLOCK_SIZE);
+    mcfgk(m * sizeof(*A));
+    for (size_t j = 0; j < k; j += BLOCK_SIZE) {     /* Loop over the columns of C */
+        for (size_t i = 0; i < n; i += BLOCK_SIZE) { /* Loop over the rows of C */
+            process_block_4x4(m, &A[i * m], m, &B[j], k, &C[(i * k) + j], k);
+        }
+    }
+}
+
+static inline void process_block_4x4(const size_t k, const float *A, const size_t lda, const float *B, const size_t ldb, float *C, const size_t ldc) {
```

Comment:
Тут небольшая опечатка в моей блочной референсной реализации :)
```suggestion
static inline void process_block_4x4(const size_t m, const float *A, const size_t lda, const float *B, const size_t ldb, float *C, const size_t ldc) {
```